poucave/html/js/app.js

```diff
@@ -698,6 +712,128 @@ class Check extends Component {
       </div>
     `;
   }
+
+  onToggleDetails(e) {
+    const { detailsOpened } = this.state;
+    this.setState({ detailsOpened: !detailsOpened });
+  }
+
+  get plotDiv() {
+    const { data } = this.props;
+    return `plot-${data.project}-${data.name}`;
+  }
+
+  togglePlot() {
+    const { result: { history = [] } } = this.props;
+    if (history.length == 0) {
+      // Data not loaded. Nothing to do.
+      return;
+    }
+
+    const { detailsOpened } = this.state;
+    if (!detailsOpened) {
+      // Details panel closed, clean-up.
+      Plotly.purge(this.plotDiv);
+      return;
+    }
+
+    // Red dots for values associated to failing statuses.
+    const failuresPlot = {
+      x: [],
+      y: [],
+      mode: 'markers',
+      marker: {
+        color: '#fa4654',
+        size: 12,
+      }
+    };
+    // Yellow lines with history of values.
+    const scalarPlot = {
+      x: [],
+      y: [],
+      mode: 'lines',
+      fill: 'tonexty',
+      fillcolor: '#ffb70030',
+      line: {
+        shape: 'hvh',
+        color: "#ffb700",
```

Comment:
I wonder if we should move all the colors here and below into constants that we can edit easily in one place?

Reply:
yeah will do, thanks!